### DISCUSSION TITLE
Is deck sync idempotent?

### DISCUSSION THREAD

**Discussion by esatterwhite** [Kong/kong#10831] [Help] [answered]
Is it safe to run `deck sync` on the old fashion DB mode frequently and in rapid succession?
We run db mode on kubernetes, but use deck to sync config - manually

My thought is to use a pre-start hook on the pod to just run it when the deployment restarts. Is this an expensive operation? can it lead to problems if 50 pods restart?

**Comment by hanshuebner** [comment, marked as answer]
The basic problem is that [`deck sync` must not be run in parallel](https://docs.konghq.com/deck/latest/faqs/#can-i-run-multiple-deck-processes-at-the-same-time).  The sync operation itself is not all that costly unless a large number of changes must be executed.  Running it with the same configuration multiple times in (quick) succession is OK and will just create database read load, so if you can serialize your runs, you should be OK.  The potential startup delay waiting for other syncs to finish may be a problem, however.

**Comment by esatterwhite** [comment]
We cannot really ensure serial executions.

**Comment by Tieske** [comment]
@esatterwhite you could maybe use this https://github.com/flant/shell-operator to watch a file, and start a sync on changes.

**Reply by esatterwhite** [reply]
It'd probably be as much work to use the ingress controller
